### ISSUE TITLE
🔥 Cannot send array/object in logEvent in v6

### ISSUE THREAD

**Issue by plaa** [invertase/react-native-firebase#2876]
## Issue

In v5 you [could provide arrays / objects](https://rnfirebase.io/docs/v5.x.x/analytics/reference/analytics#logEvent) within the `logEvent` `params` object, and such usage is even provided in [Firebase's own examples](https://firebase.google.com/docs/analytics/events?platform=web#log_events_3).

In v6 `logEvent` is [restricted to having strings, numbers and booleans](https://invertase.io/oss/react-native-firebase/v6/analytics/reference/module#logEvent) as values and it throws an exception if any other type is provided:

![Screenshot of error](https://user-images.githubusercontent.com/1436281/68856659-bf9b6400-06e9-11ea-8f90-7cb63a0f9b20.png)

It seems this is a major limitation in v6 compared to what Firebase analytics offers.

_(Edit: initially wrote getEvent instead of logEvent, still wrong in the title)_

---

### iOS

<details><summary>Click To Expand</summary>
<p>

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

```ruby
# N/A
```

#### `AppDelegate.m`:

```objc
// N/A
```

</p>
</details>

---

### Android

<details><summary>Click To Expand</summary>
<p>

#### Have you converted to AndroidX?

- [ ] my application is an AndroidX application?
- [ ] I am using `android/gradle.settings` `jetifier=true` for Android compatibility?
- [ ] I am using the NPM package `jetifier` for react-native compatibility?

#### `android/build.gradle`:

```groovy
// N/A
```

#### `android/app/build.gradle`:

```groovy
// N/A
```

#### `android/settings.gradle`:

```groovy
// N/A
```

#### `MainApplication.java`:

```java
// N/A
```

#### `AndroidManifest.xml`:

```xml
<!-- N/A -->
```

</p>
</details>

---

## Environment

<details><summary>Click To Expand</summary>
<p>

**`react-native info` output:**

```
info
  React Native Environment Info:
    System:
      OS: macOS 10.14.6
      CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
      Memory: 149.97 MB / 32.00 GB
      Shell: 5.0.9 - /usr/local/bin/bash
    Binaries:
      Node: 12.8.1 - /usr/local/bin/node
      Yarn: 1.17.3 - /usr/local/bin/yarn
      npm: 6.11.3 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 13.1, DriverKit 19.0, macOS 10.15, tvOS 13.0, watchOS 6.0
    IDEs:
      Android Studio: 3.5 AI-191.8026.42.35.5791312
      Xcode: 11.1/11A1027 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.9.0 => 16.8.3
      react-native: 0.61.4 => 0.59.9
```

- **Platform that you're experiencing the issue on**:
  - [ ] iOS
  - [ ] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [x] Both
- **`react-native-firebase` version you're using that has this issue:**
  - `6.0.3`
- **`Firebase` module(s) you're using that has the issue:**
  - `Analytics`
- **Are you using `TypeScript`?**
  - `Y` & `3.7.2`
  
</p>
</details>

---

Think `react-native-firebase` is great? Please consider supporting all of the project maintainers and contributors by donating via our [Open Collective](https://opencollective.com/react-native-firebase/donate) where all contributors can submit expenses. [[Learn More]](https://invertase.io/oss/react-native-firebase/contributing/donations-expenses)

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

**Comment by Ehesp** [comment]
Hmm I'm not sure why that's been implemented like that. From memory one of the platforms didn't accept nested data... I'll take a look into it.

**Comment by plaa** [comment]
Would still appreciate knowing whether this will be implemented or not.

**Comment by Dylan0916** [comment]
> Would still appreciate knowing whether this will be implemented or not.

+1

**Comment by josephtactica** [comment]
+1

**Comment by josephtactica** [comment]
> Would still appreciate knowing whether this will be implemented or not.

Same here.

**Comment by stereodenis** [comment]
same here

**Comment by josephtactica** [comment]
I've used with `params = [...]`

`const logger = async (name, params) => {
  await analytics().native.logEvent(name, params);
};`

**Comment by plaa** [comment]
Still waiting on whether this will be implemented or whether we need some
workarounds.

**Comment by Salakar** [comment]
@plaa and others, @russellwheatley has fixed this in #3219 which has been merged - this will be available in a upcoming release, thanks
